examples/java/README.md

```diff
@@ -0,0 +1,80 @@
+since there is no "eval" function in java.
+
+Synchronous/asynchronous:
+The plugin is partly asynchronous since several method calls can be made to the plugin at the same time. But a `StreamObserver` (which sends data to and from the plugin) can only handle one call at a time, and they need 
+to be in a specific order, therefore the plugin cannot be fully asynchronous.
+
+## Prerequisites
+This example was created using maven. There are other ways to do this, what you need is the protocol buffer package and the grpc package (in that order) and the protoc 
+compiler to generate the code from the .proto file. You also need Java 8.
+
+If you want to run the example using maven, use the standard maven source tree (and make sure you have installed [maven](http://maven.apache.org/ )). 
+Start with putting the `pom.xml` file in a directory somewhere (from now on called `my-app`). In the `my-app` directory create a folder called `src`. 
+Place the `ServerSideExtension.proto` file both in `my-app/src/main/proto` and in `my-app/src/test/proto`. The `PluginServer.java` file should be placed in `my-app/src/main/java/Plugin` 
+and the `AppTest.java` file should be placed in `my-app/src/test/java/Plugin`.
+
+
+## Generate the java plugin (and the classes from the .proto file)
+**In a command prompt:** go to the `my-app` directory and run `mvn package`.
+
+Now you can see a new folder called `target` in the `my-app` directory. In the `target` folder, you can find a jar file, that is the plugin. The code generated from the `.proto` file can be found 
+in the `generated-sources` folder.
+
+### If you are having trouble...
+It is possible to install the protobuff and the grpc packages manually if they for some reason are not downloaded by maven when you run `mvn package`. If you need to do this, install the protocol buffer first
+since grpc is dependent on the protocol buffer. 
+
+#### Install Protocol buffer
+Download a protoc compiler (protoc-[version]-[Platform].zip) and and the protocol buffer for java zip or tar.gz (protobuf-java-[version]-zip/tar.gz) from the 
+[release page for protbuf](https://github.com/google/protobuf/releases). Make sure the version number is the same. Extract the files, go to the java directory and follow the instructions 
+in the [README](https://github.com/google/protobuf/blob/master/java/README.md). You have already done half of step two.
+
+#### Install Grpc
+Download [grpc-java](https://github.com/grpc/grpc-java) and in the grpc-java directory create a `gradle.properties` file with the text `skipCodegen=true`. 
+
+**In a command prompt:** go to the grpc-java directory, run `gradlew.bat build` and `gradlew.bat install`. This installs grpc in your local maven repository (usually located at `C:/Users/<user>/.m2`).
+
+## Changing the default logging
+Move the `javapluginlogger.properties` file to the `my-app/target` directory (it needs to be in the same directory as the `my-app-1.0-SNAPSHOT.jar` file).
```

Comment:
I don't know maven, but I suspect that this file could be moved automatically during the build?